{{ cookiecutter.github_repo_name }}/src/{{ cookiecutter.package_dir_name }}/calculator.py

```diff
@@ -2,5 +2,42 @@
 
 
 def dot_product(a, b):
-    """Calculate the dot product of two vectors."""
-    return np.dot(a, b)
+    """Compute the dot product of two vectors of any size.
+
+    Ensure that the inputs of a and b are of the same size.
+    The supported types are Python list, Python tuple, and NumyPy array.
+
+    Parameters
+    ----------
+    a : array_like
+        The first input of vector.
```

Comment:
I think there is a rogue "of" in this line.

Reply:
fixed

Reply:
`numpy.dot` refers to the type as `array_like` https://numpy.org/doc/stable/reference/generated/numpy.dot.html. I have added `...which can be converted to a NumPy array. Examples include lists and tuples.`